src/olympia/addons/serializers.py

```diff
@@ -300,7 +300,7 @@ def fake_version_object(self, obj, data):
             version = Version(
                 addon=obj, id=data['id'],
                 reviewed=self.handle_date(data['reviewed']),
-                version=data['version'])
+                version=data['version'], channel=data['channel'])
```

Comment:
Ah. If you want to avoid the indexing stuff: pass a `channel` argument to `fake_version_object()`. We only use it for `obj.current_beta_version` (has to be listed), `obj._current_version` (has to be listed) and `obj.latest_unlisted_version` (has to be unlisted).

Also add a test to make sure `url` is correct in the serializer output for all cases.